### PR TITLE
Out of curiosity, how backward compatible are things?

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Yesterday I used `kr8s==0.20.6` (which supports Kubernetes `1.30+`) on a Kubernetes `1.26` cluster and things mostly worked, but there were a few weird edge cases.

Out of curiosity I'm interested to see how far back we can go and have tests still pass.

**This does not signal our intention to change [the way we calculate which versions we support](https://docs.kr8s.org/en/latest/installation.html#supported-kubernetes-versions). I'm just curious and the easiest way to test it out is to open a PR and let CI run.**

### CODE REVIEW

File: kr8s/tests/test_objects.py [hunk 1]
```diff
@@ -588,6 +588,21 @@ async def test_all_v1_objects_represented():
         assert get_class(obj["kind"], obj["version"])
 
 
+@pytest.mark.parametrize(
+    "name, object_class",
+    [
+        ("pods", Pod),
+        ("services", Service),
+        ("configmaps", ConfigMap),
+        ("nodes", Node),
```
Comment (jacobtomlinson) [inline kr8s/tests/test_objects.py:597]: This is what I was seeing fail on `1.26`. When running `kr8s.get("nodes")` it was constructing a object new class instead of using `Node`.

**Comment by jacobtomlinson** [comment]
As expected tests are failing on `1.26` and below.
